### ISSUE TITLE
fetchzip does not work correctly with unicode file names

### ISSUE THREAD

**Issue by piegamesde** [NixOS/nixpkgs#118612]
**Describe the bug**
When you have a zip file, and it contains files which have unicode characters in their name, you'll get the wrong results.

**To Reproduce**
1. Get a file with unicode characters in their file name, for example like https://extensions.gnome.org/extension-data/flypieschneegans.github.com.v4.shell-extension.zip
2. Prefetch it `nix-prefetch-url --unpack https://extensions.gnome.org/extension-data/flypieschneegans.github.com.v4.shell-extension.zip`
3. Download it using `fetchzip`: `nix-build -E 'let pkgs = import ./default.nix {}; in pkgs.fetchzip { url = "https://extensions.gnome.org/extension-data/flypieschneegans.github.com.v4.shell-extension.zip"; sha256 = "15pkdslpyvmpn4dfgmz89nl63b1jk9xbb9dpfp9xffdkkir95kgy";stripRoot = false;}'`
4. Diff them: `diff -r /nix/store/h0xbzms5y4jyawcx0qxakiaql13cycwq-flypieschneegans.github.com.v4.shell-extension.zip /nix/store/mr1qgxb44kb79xw9rd7b44gc0nff00qm-source`

**Expected behavior**
Both are identical.

**Actual behavior**

```
Only in /nix/store/h0xbzms5y4jyawcx0qxakiaql13cycwq-flypieschneegans.github.com.v4.shell-extension.zip/presets: ⚪ Default Bright.json
Only in /nix/store/h0xbzms5y4jyawcx0qxakiaql13cycwq-flypieschneegans.github.com.v4.shell-extension.zip/presets: ⚫ Default Dark.json
Only in /nix/store/mr1qgxb44kb79xw9rd7b44gc0nff00qm-source/presets: #U26aa Default Bright.json
Only in /nix/store/mr1qgxb44kb79xw9rd7b44gc0nff00qm-source/presets: #U26ab#Ufe0f Default Dark.json
```

You can see that the '⚪' symbol got replaced with '#U26aa' for some reason.

**Notify maintainers**
Who maintains `fetchzip`? Pinging previous contributors at random: @edolstra @oxij @MetaDark @Ericson2314 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.8-arch1-1, Arch Linux, noversion`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels: `"nixpkgs-20.09.3686.da7f4c48425, nixpkgs-unstable-21.05pre278688.c0e88185200"`
 - nixpkgs: `~/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by maydayv7** [comment]
(Bumping issue, as am affected)

**Comment by piegamesde** [comment]
Oh wow, I didn't think this would ever happen to someone else. I guess the law of big numbers has struck again.

Could you please add your example to the list?

**Comment by maydayv7** [comment]
Well, actually it's for the same (`gnomeExtensions.fly-pie`)

**Comment by Zimmi48** [comment]
We have also observed an issue related to Unicode in file names with `fetchzip` but it only affected Darwin. See: https://github.com/NixOS/nixpkgs/pull/138274#issuecomment-933322839

**Comment by LeSuisse** [comment]
This was solved thanks to #176253
